sdk/cpp/core/src/gnmi_client.cpp

```diff
@@ -1,273 +1,325 @@
+    client_cert.assign((std::istreambuf_iterator<char>(cf)),(std::istreambuf_iterator<char>()));
+    ssl_opts = {server_cert, client_key, client_cert};
+    */
+
+    auto channel_creds = grpc::SslCredentials(grpc::SslCredentialsOptions(ssl_opts));
+    grpc::CreateCustomChannel(address, channel_creds, args);
+    get_capabilities();
+    return EXIT_SUCCESS;
 }
 
-int gNMIClient::connect() 
+static bool check_capabilities_status(Status status) 
 {
-    grpc::CreateChannel("0.0.0.0:50051", grpc::InsecureChannelCredentials());
-    init_capabilities();
-    return EXIT_SUCCESS;
+    if (!(status.ok())) 
+    {
+      YLOG_ERROR("Capabilities RPC Failed.");
+      throw(YCPPError{"Capabilities RPC Failed"});
+      return false;
+    }
+    return true;
 }
 
-void gNMIClient::init_capabilities() 
+bool gNMIClient::has_gnmi_version(::gnmi::CapabilityResponse* response) 
 {
-	std::string cap;
-	while (std::getline(capfile, cap)) 
-	{
-		capabilities.push_back(cap);
-		//std::cout << cap;
-	}
+    if (!(response->gnmi_version().size() > 0)) 
+    {
+        YLOG_ERROR("Capabilities Received Without gNMI Version");
+        throw(YCPPError{"Capabilities Received Without gNMI Version"});
+        return false;
+    }
+    return true; 
 }
 
-bool gNMIClient::get_capabilities(const CapabilityRequest& request, CapabilityResponse* response) 
+void gNMIClient::parse_capabilities_modeldata(::gnmi::CapabilityResponse* response)
 {
-	grpc::ClientContext context;
-	grpc::Status status = stub_->Capabilities(&context, request, response);
-	if (!status.ok()) 
-	{
-	  std::cout << "Capabilities RPC Failed." << std::endl;
-	  return false;
-	}
-	if (!response->gnmi_version().size() > 0) 
-	{
-	  std::cout << "Server has not returned gnmi version" << std::endl;
-	  return false;
-	} else 
-	{
-		std::cout << "Receive capability response with gNMI version: " << response->gnmi_version() << std::endl;
-		for (unsigned int i = 0, n = response->supported_models_size(); i < n; i++) 
-		{
-			std::cout << "supported_models size" << response->supported_models_size() << std::endl;
-			std::cout << "supported_models";
-			response->supported_models(i);
-		}
-		for (int i = 0, n = response->supported_encodings_size(); i < n; i++) 
-		{
-			std::cout << "supported_models size" << response->supported_encodings_size() << std::endl;
-			std::cout << "supported_models";
-			response->supported_encodings(i);
-		}
-	}
-    return true;
+    ::gnmi::ModelData modeldata;
+    string cap, name, organization, version; 
+
+    for (int i = 0, n = response->supported_models_size(); i < n; i++) 
+    {
+        cap.clear();
+        modeldata = response->supported_models(i);
+        name = modeldata.name();
+        organization = modeldata.organization();
+        version = modeldata.version();
+        if(!(modeldata.name().empty()))
+        {
+            YLOG_INFO("Name: {}", name.c_str());
```

Comment:
`YLOG_INFO` is meant to be reserved for user-level messages (only for things like payload sent/received or connection established/closed). So, if this message is mainly for developers/debugging purpose, perhaps `YLOG_DEBUG` could be better to use here?